src/http/server.ts

```diff
@@ -565,6 +565,47 @@ export class Server {
           ),
         };
 
+        newPaths.push(pathObj);
+      } else if (path.includes("?") === true) { // optional params
+        let tmpPath = path;
+        // Replace required params
+        const numberOfRequiredParams = path.split("/").filter((param) => {
+          return (param.includes(":") || param.includes("{")) &&
+            !param.includes("?");
+        }).length;
+        for (let i = 0; i < numberOfRequiredParams; i++) {
+          tmpPath = tmpPath.replace(
+            /(:[^(/]+|{[^0-9][^}]*})/, // same as REGEX_URI_MATCHES but not global
+            Server.REGEX_URI_REPLACEMENT,
+          );
+        }
+        // Replace optional path params
+        const maxOptionalParams = path.split("/").filter((param) => {
+          return param.includes("?");
+        }).length;
+        for (let i = 0; i < maxOptionalParams; i++) {
+          if (i === 0) { // We need to mark the start for the first optional param
+            tmpPath = tmpPath.replace(
```

Comment:
could you add a comment about what's going on here

Reply:
https://github.com/drashland/deno-drash/pull/337/commits/f5c5cd2c65f03b573428eff9d0760f9b3a487389